hack/csv-generator.go

```diff
@@ -0,0 +1,263 @@
+	gyaml "github.com/ghodss/yaml"
+	csvv1 "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/spf13/cobra"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	extv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+type generatorFlags struct {
+	dumpCRDs          bool
+	csvVersion        string
+	namespace         string
+	validatorImage    string
+	kvmInfoImage      string
+	virtLauncher      string
+	nodeLabellerImage string
+	cpuPlugin         string
+	operatorImage     string
+}
+
+type csvClusterPermissions struct {
+	ServiceAccountName string              `json:"serviceAccountName"`
+	Rules              []rbacv1.PolicyRule `json:"rules"`
+}
+
+type csvPermissions struct {
+	ServiceAccountName string              `json:"serviceAccountName"`
+	Rules              []rbacv1.PolicyRule `json:"rules"`
+}
+
+type csvDeployments struct {
+	Name string                `json:"name"`
+	Spec appsv1.DeploymentSpec `json:"spec,omitempty"`
+}
+
+type csvStrategySpec struct {
+	ClusterPermissions []csvClusterPermissions `json:"clusterPermissions"`
+	Permissions        []csvPermissions        `json:"permissions"`
+	Deployments        []csvDeployments        `json:"deployments"`
+}
+
+var (
+	f       generatorFlags
+	rootCmd = &cobra.Command{
+		Use:   "csv-generator",
+		Short: "csv-generator for ssp operator",
+		Long:  `csv-generator generates deploy manifest for ssp operator`,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := runGenerator()
+			if err != nil {
+				fmt.Println(err.Error())
+				os.Exit(1)
+			}
+		},
+	}
+)
+
+func main() {
+	rootCmd.Execute()
+}
+
+func init() {
+	rootCmd.Flags().StringVar(&f.csvVersion, "csv-version", "", "Version of csv manifest (required)")
+	rootCmd.Flags().StringVar(&f.namespace, "namespace", "", "Namespace in which ssp operator will be deployed (required)")
+	rootCmd.Flags().StringVar(&f.operatorImage, "operator-image", "", "Link to operator image (required)")
+	rootCmd.Flags().StringVar(&f.validatorImage, "validator-image", "", "Link to template-validator image")
+	rootCmd.Flags().StringVar(&f.nodeLabellerImage, "node-labeller-image", "", "Link to node-labeller image")
+	rootCmd.Flags().StringVar(&f.kvmInfoImage, "kvm-info-image", "", "Link to kvm-info-nfd-plugin image")
+	rootCmd.Flags().StringVar(&f.virtLauncher, "virt-launcher-image", "", "Link to virt-launcher image")
+	rootCmd.Flags().StringVar(&f.cpuPlugin, "cpu-plugin-image", "", "Link to cpu-nfd-plugin image")
+	rootCmd.Flags().BoolVar(&f.dumpCRDs, "dump-crds", false, "Dump crds to stdout")
+
+	rootCmd.MarkFlagRequired("csv-version")
+	rootCmd.MarkFlagRequired("namespace")
+	rootCmd.MarkFlagRequired("operator-image")
+}
+
+func runGenerator() error {
+	csvFile, err := ioutil.ReadFile("/data/olm-catalog/ssp-operator.clusterserviceversion.yaml")
```

Comment:
The path to the `data` directory should be relative and not obsolete (that's how common-templates use it)

Reply:
`manager` binary is located in `/` but csv-generator is located in `/usr/bin/` so I have to use absolute path

Reply:
So let's please align both of them to be in `/`